iceoryx_examples/iceoptions/iox_publisher_with_options.cpp

```diff
@@ -51,24 +51,33 @@ int main()
     publisherOptions.subscriberTooSlowPolicy = iox::popo::ConsumerTooSlowPolicy::WAIT_FOR_CONSUMER;
     //! [too slow policy]
 
+    //! [create publisher with options]
     iox::popo::Publisher<RadarObject> publisher({"Radar", "FrontLeft", "Object"}, publisherOptions);
+    //! [create publisher with options]
 
     // we have to explicitely offer the publisher for making it visible to subscribers
     //! [offer]
     publisher.offer();
     //! [offer]
 
     double ct = 0.0;
-    while (!iox::posix::hasTerminationRequested())
-    {
-        ++ct;
 
-        // Retrieve a sample, construct it with the given arguments and publish it via a lambda.
-        publisher.loan(ct, ct, ct).and_then([](auto& sample) { sample.publish(); });
+    std::thread publishData([&]() {
+        while (!iox::posix::hasTerminationRequested())
+        {
+            ++ct;
+
+            // Retrieve a sample, construct it with the given arguments and publish it via a lambda.
+            publisher.loan(ct, ct, ct).and_then([](auto& sample) { sample.publish(); });
+
+            std::cout << APP_NAME << " sent value: " << ct << std::endl;
 
-        std::cout << APP_NAME << " sent value: " << ct << std::endl;
+            std::this_thread::sleep_for(std::chrono::milliseconds(400));
+        }
+    });
 
-        std::this_thread::sleep_for(std::chrono::milliseconds(400));
+    while (!iox::posix::hasTerminationRequested())
```

Comment:
Could you please replace this loop with `iox::posix::waitForTerminationRequest();` then this is perfect an can be merged.

Reply:
Unfortunately not, because than RouDi unblocks the shutdown of this app immediately and the subscriber receives no data any longer.

Reply:
Sorry, ignore what I wrote. It's too late, it was a c&p error.